Create unified Bitcoin design principles page (#144)
BitcoinDesign/Guide: guide/principles/principles.md

---
layout: guide

> title: Bitcoin design principles

Author: I had it like that first. Then the page itself felt like the title made more sense as 'Bitcoin Design Principles' to make it clear without additional context what the page is about. 

Changed the nav title to match, but perhaps it's nicer as just Principles, and the extra context is not needed there.

> title: Bitcoin design principles

Author: Changed back. 0f557c7

description: The key principles to follow when designing Bitcoin products.
nav_order: 2
has_children: false

> permalink: /guide/principles/principles/

Reviewer: ```suggestion
permalink: /guide/principles/
```

Don't need the extra `/principles`.

> permalink: /guide/principles/principles/

Author: Fixed, removed the extra /principles

main_classes: -no-top-padding
---

<!--

Editor's notes

The key principles of designing for Bitcoin

-->

# Bitcoin design principles

> As a new technology, Bitcoin offers the opportunity of a decentralized open financial system without intermediaries. To make this a reality, we encourage everyone that is working on products to deliberately support the core principles of designing for Bitcoin. Some of these come from the technology itself, and others from the behaviour and ethos of the community. Although every use case and product is different, all applications should strive to follow these principles. Diverging from them should only be done with very good reason.

Reviewer: How about adding that these are principles that we have identified as a community? Not sure we should impose absolute views here.

Author: Added. 62d7a7a

> As a new technology, Bitcoin offers the opportunity of a decentralized open financial system without intermediaries. To make this a reality, we encourage everyone that is working on products to deliberately support the core principles of designing for Bitcoin. Some of these come from the technology itself, and others from the behaviour and ethos of the community. Although every use case and product is different, all applications should strive to follow these principles. Diverging from them should only be done with very good reason.

Author: Applied in f29f2d7

- [Decentralization](#decentralization)
- [Inclusion](#inclusion)
- [Transparency](#transparency)
- [Self-custody](#self-custody)
- [Security](#security)
- [Privacy](#privacy)
- [Interoperability](#interoperability)

---

## Decentralization

> Unlike traditional banking systems, the Bitcoin economy does not require new users to seek permission from anyone. Bitcoin has no central point of control. No one person or entity is in charge. Connecting to any node on the network gives you the same rights and responsibilities, which also ensures that there is no single point of failure.

Reviewer: There might be an extra point in here that decentralization is only possible when product makers pro-actively build that way. So they should ensure that projects can continue without their input, users can port their funds to other projects easily, etc. Decentralization goes pretty deep when you want to do it right.

Author: Added a 'Don't - Build products that stop working if the project shuts down'. 6fbc9ac

> *Build products that are permissive in terms of what users can do with Bitcoin.* <br>

Reviewer: I like these italic summaries. You think they could go right below the title of each principle as a one-sentence summary?

Author: Good idea. Reorganized and for now formatted with block quotes. df4d515

*Have fall-back options for connections to the network.*

**Do**
- Manage your own private keys
- Run a full Bitcoin node when possible
- Alternatively, use a light client with the [p2p network]({{ 'https://github.com/bitcoin/bips/blob/master/bip-0157.mediawiki'}}) using [compact block filters]({{ 'https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki'}}) 

> - Offer the user choice of what node to connect to

Reviewer: Wonder if this is a larger point than just nodes? Should I be able to choose an explorer, or where to get currency conversion data from? All external connections can leak meta data.

Author: Added 'and other external services'. 7a7d5c2

**Don't**
- Introduce a single point of failure between the user and the Bitcoin network

---

## Inclusion

There are no background checks, credit checks, or gatekeepers to Bitcoin. A Kenyan farmer has the same access to Bitcoin as a Wall Street trader. 

> And while Bitcoin is already widely used by a large number of people, it pales in comparison with the billions that will use it in the future. Without dumbing down or reducing the possibilities we need to design products that are prepared for people that are new to Bitcoin. This means using plain and ordinary language, explaining things in the context where they are needed, not overwhelming people with technical detail, and more.

Reviewer: ```suggestion
And while Bitcoin is already widely used by a large number of people, it pales in comparison with the billions that will use it in the future. We need to design products that are prepared for people new to Bitcoin, without reducing the possibilities. This means using plain and familiar language, explaining things in the context where they are needed, not overwhelming people with technical detail, and more.
```

Removed `dumbing down`, as it's not a clear idea.

> And while Bitcoin is already widely used by a large number of people, it pales in comparison with the billions that will use it in the future. Without dumbing down or reducing the possibilities we need to design products that are prepared for people that are new to Bitcoin. This means using plain and ordinary language, explaining things in the context where they are needed, not overwhelming people with technical detail, and more.

Author: Toned down. 5e3906d

*Provide equal and direct access to the Bitcoin network, without middlemen and exclusive requirements.* <br>
*Design Bitcoin products that are usable by the widest range of people possible.*

**Do**
- Use plain language that people new to Bitcoin can understand regardless of prior knowledge
- Use more languages than just english
- Educate in place, when people are presented with a new concept
- Treat users with impairments, or who rely on assistive technologies, as first-class citizens

**Don't**
- Introduce requirements that might exclude many people, for example by building only for high-end hardware, requiring US KYC (Know-your-customer) information etc

> - Avoid technical detail that is not required knowledge, or technical terms like seed phrase XPUBs, mnemonics etc.

Reviewer: ```suggestion
- Add technical detail that is not required knowledge, or technical terms like seed phrase XPUBs, mnemonics etc.
```

It's in `Don't`, so no need for the negative statement.

> - Do not put all education up front and expect people to read and remember it

Reviewer: ```suggestion
- Put all education up front and expect people to read and remember it
```

It's in `Don't`, so no need for the negative statement.

> - Do not put all education up front and expect people to read and remember it

Author: I think it's used correctly as 'up front' in this context. 
https://grammarist.com/usage/upfront-up-front/

---

## Transparency

While an open and decentralized financial system that lets people connect directly with it is great, it does put a lot of burden on the users to pick a product they trust, and that works for them. We can make this easier by freely sharing information about how our products work, what technologies they use and rely on and so forth. Let people verify that your claims are true. It is also very important to be open about what risks the user might be taking on with self-custody, under which scenarios they might lose access to their funds and best practices for avoiding this. 

> *Be open and transparent with how your product works.* <br>

Reviewer: Should these be under `Do`?

Author: Added. 6bf33a0
(Will fix duplication with italic text in upcoming commit.)

*Explain what risks the user is taking on, and how best to mitigate them.*

**Do**

> - Build your product as open source when possible

Reviewer: ```suggestion
- Build your product as open-source when possible
```

- Let people verify, without having to trust
- Be clear about what risk the user is taking on, and how they can lose access to their funds 

**Don't**
- Make claims that are not explained or verifiable 

---

## Self-custody

Our existing mental models of access to digital services are usernames and passwords, controlled by a company that has custody of your funds and data. With everyone having direct access to the Bitcoin network without middlemen, there is no longer a need for people to delegate control of their funds. While it comes with greater responsibility, it is self-custody that really enables the open financial system of peer-to-peer transactions.

*Give end users full control of their funds.*

**Do**
- Let users control their private keys, and funds directly

**Don't**
- Custody funds for your users

---

> ## Security

Reviewer: You think it's worth noting that security is also a feeling? My wallet can have the best private key management but I might still not feel secure when it looks crappy, or there's something not clear about the backup mechanisms.

Author: Added. 7b8e2f3

> Self-custody means that the end user is ultimately responsible for keeping their funds and private keys safe. They can only do that if we provide them with good, appropriate tools and awareness of good practice. It is therefore our responsibility as designers and makers to really take their security seriously and make Bitcoin products that are as secure as possible by default. We should strive for no loss of funds, regardless of whether this happens through negligence or theft. 

Author: Changed to 'best practices'. 256e0ae

Security is especially important when onboarding people new to Bitcoin where solutions that might be fine for advanced users might backfire for beginners. New users are likely to start by storing small amounts until they get comfortable with self-custody. A scheme suited for storing large amounts of funds is likely overkill and scary to a beginner, so the concept of progressive security is a good idea. This could let a user upgrade their security and private key management scheme as their savings grow. 

> A big part of security is also education and awareness, protecting users from bad actors and potentially their own bad behavior. Bad actors may already know what most users don’t, and be aware of new vulnerabilities. It is unrealistic to expect beginners to take onboard all the knowledge acquired by advanced users of several years in one go, for example while onboarding to a product. We should therefore consider how to continuously educate and level up user awareness of best practices and risks.

Reviewer: ```suggestion
A big part of security is also education and awareness, protecting users from bad actors and potentially their own bad behavior. Bad actors may already know what most users don’t, and be aware of new vulnerabilities. It is unrealistic to expect beginners to take in all the knowledge acquired by advanced users of several years in one go, for example while onboarding to a product. We should therefore consider how to continuously educate and level up user awareness of best practices and risks.
```

Avoiding the term `onboard` twice in one sentence.

---

<!--

Editor's notes

The key principles of designing for Bitcoin

-->

# Bitcoin design principles

As a new technology, Bitcoin offers the opportunity of a decentralized open financial system, where participants share the role of securing the network. This is important to give everyone equal and direct access without needing intermediaries. To make this a reality, we encourage everyone working on products to deliberately support the core principles of designing for Bitcoin. These are principles we in the Bitcoin Design Community have identified and stand behind. Some of these come from the technology itself, and others from the behaviour and ethos of the community. Although every use case and product is different, applications should strive to follow these principles. Diverging from them should only be done with very good reason.

- [Decentralization](#decentralization)
- [Inclusion](#inclusion)
- [Transparency](#transparency)
- [Self-custody](#self-custody)
- [Security](#security)
- [Privacy](#privacy)
- [Interoperability](#interoperability)

---

## Decentralization

> Encourage users to run their own Bitcoin node and connect directly to the network

> Unlike traditional banking systems, the Bitcoin economy does not require new users to seek permission from anyone. Bitcoin has no central point of control. No one person or entity is in charge. Connecting to any [node]({{ '/guide/glossary/#node/' | relative_url }}) on the network gives you the same rights and responsibilities, which also ensures that there is no single point of failure.

Reviewer 2: If append `/` at the end of the link it doesn't redirects to the header. this fixes it.
```suggestion
Unlike traditional banking systems, the Bitcoin economy does not require new users to seek permission from anyone. Bitcoin has no central point of control. No one person or entity is in charge. Connecting to any [node]({{ '/guide/glossary/#node' | relative_url }}) on the network gives you the same rights and responsibilities, which also ensures that there is no single point of failure.
```

- Alternatively, use a light client with the [p2p network]({{ 'https://github.com/bitcoin/bips/blob/master/bip-0157.mediawiki'}}) using [compact block filters]({{ 'https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki'}}) 
- Offer the user choice of what node and other external services to connect to

**Don't**
- Introduce a single point of failure between the user and the Bitcoin network
- Build products that stop working if the project shuts down

---

## Inclusion

> Provide equal access regardless of the user's location, language, ability and prior knowledge of Bitcoin

There are no background checks, credit checks, or gatekeepers to Bitcoin. A Kenyan farmer has the same access to Bitcoin as a Wall Street trader. 

And while Bitcoin is already widely used by a large number of people, it pales in comparison with the many more that are likely to use it in the future. We need to design products that are prepared for people new to Bitcoin, without reducing the possibilities. This means using plain and familiar language, explaining things in the context where they are needed, not overwhelming people with technical detail, and more.

**Do**

> - Provide equal and direct access to the Bitcoin network, without middlemen and exclusive requirements

Reviewer 2: > Provide equal and direct access to the Bitcoin network, without middlemen and exclusive requirements

I'd try to break this down somehow and simplify it, I'm not sure it's clear to a newbie what we mean by this?

Author: Addressed in 661e72f

- Design Bitcoin products that are usable by the widest range of people possible
- Use plain language that people new to Bitcoin can understand regardless of prior knowledge
- Use more languages than just English
- Educate in place, when people are presented with a new concept
- Treat users with impairments, or who rely on assistive technologies, as first-class citizens

**Don't**

> - Introduce requirements that might exclude many people, for example by building only for high-end hardware, requiring US KYC (Know-your-customer) information etc

Author: Addressed in 17035b9

- Add technical detail that is not required knowledge, or technical terms like seed phrase XPUBs, mnemonics etc.
- Put all education up front and expect people to read and remember it

---

## Transparency

> Open-source your products and clearly explain how they work

While an open and decentralized financial system that lets people connect directly with it is great, it does put a lot of burden on the users to pick a product they trust, and that works for them. We can make this easier by freely sharing information about how our products work, what technologies they use and rely on and so forth. Let people verify that your claims are true. It is also very important to be open about what risks the user might be taking on with self-custody, under which scenarios they might lose access to their funds and best practices for avoiding this. 

**Do**
- Be open and transparent with how your product works
- Build your product as open-source when possible

> - Let people verify, without having to trust

Reviewer 2: I think we should elaborate on the `Don't trust - verify` a bit more in the opening paragraph. The bullet-point says basically the same thing as the one in the paragraph above.

Author: Addressed in 661e72f

- Explain what risks the user is taking on, and how best to mitigate them

**Don't**
- Make claims that are not explained or verifiable 

---

## Self-custody

> Let users control their private keys, with no risk for seizure or freezing of funds

Our existing mental models of access to digital services are usernames and passwords, controlled by a company that has custody of your funds and data. With everyone having direct access to the Bitcoin network without middlemen, there is no longer a need for people to delegate control of their funds. While it comes with greater responsibility, it is self-custody that enables the open financial system of peer-to-peer transactions.

**Do**
- Let users control their private keys and funds directly

**Don't**
- Custody funds for your users
- Build products where the users' funds can be seized, or frozen

---

## Security

> Provide appropriate and progressive security for all types of users

> Self-custody means that the end user is ultimately responsible for keeping their funds and private keys safe. They can only do that if we provide them with good, appropriate tools and awareness of best practices. It is therefore our responsibility as designers and makers to really take their security seriously and make Bitcoin products that are as secure as possible by default. We should strive for no loss of funds, regardless of whether this happens through negligence or theft. 

Reviewer 2: Starts talking about self-custody, but we should be concise and focus on security as we've talked about the self-custody above.

Author: Addressed in bbd6299

> Security is especially important when onboarding people new to Bitcoin where solutions that might be fine for advanced users might backfire for beginners. New users are likely to start by storing small amounts until they get comfortable with self-custody. A scheme suited for storing large amounts of funds is likely overkill and scary to a beginner, so the concept of progressive security is a good idea. This could let a user upgrade their security and private key management scheme as their savings grow. 

Reviewer 2: > Security is especially important when onboarding people new to Bitcoin

Would be good to cross-link to `onboarding/introduction/`

Author: Addressed in 7cfbcba

A big part of security is also education and awareness, protecting users from bad actors and potentially their own bad behavior. Bad actors may already know what most users don’t, and be aware of new vulnerabilities. It is unrealistic to expect beginners to take in all the knowledge acquired by advanced users of several years in one go, for example while onboarding to a product. We should therefore consider how to continuously educate and level up user awareness of best practices and risks.

Security can also be a feeling. A product that is polished, good looking, easy to use and transparently communicates how it works can help users feel more secure. Especialy compared to one with the exact same security measures that lack these qualities.

**Do**

> - Take safeguarding of users funds seriously

Reviewer 2: - Provide suitable private key management schemes for beginners

I feel like this one was already mentioned in the paragraphs above. Maybe you had something different in mind, what do you mean when you say `sustainable PKM`?

Author: The do's and don'ts sometimes repeat information from the paragraph in bullet points, I think that's OK.

Sustainable -> Suitable (meaning appropriate for beginners)

---
layout: guide
title: Principles
description: The key principles to follow when designing Bitcoin products.
nav_order: 2
has_children: false
permalink: /guide/principles/principles/
main_classes: -no-top-padding
---

<!--

Editor's notes

The key principles of designing for Bitcoin

-->

# Bitcoin Design Principles

> As a new technology, Bitcoin offers the opportunity of an open financial system without intermediaries. To make this a reality, we encourage everyone that is working on products to deliberately support the core principles of designing for Bitcoin. Some of these come from the technology itself, and others from the behaviour and ethos of the community. Although every use case and product is different, all applications should strive to follow these principles. Diverging from them should only be done with very good reason.

Reviewer 2: > As a new technology, Bitcoin offers the opportunity of an open financial system without intermediaries. To make this a reality, we encourage everyone that is working on products to deliberately support the core principles of designing for Bitcoin.

The intro sounds more related to "Permissionless" than "Decentralization", perhaps another principles, is it worth distinguishing the two?

Author: Good point. I'd like to combine the two if possible.
I see the permission-less aspect as a result of the decentralization, but perhaps it can be expressed better.
Will add decentralized in the first sentence for now.

Reviewer 3: I also think why `an open financial system without intermediaries` is important is not immediately obvious, maybe we just say it's available for everyone but link the trust minimized nature to decentralization

Author: Addressed in 3b98813

Security can also be a feeling. A product that is polished, good looking, easy to use and transparently communicates how it works can help users feel more secure. Especialy compared to one with the exact same security measures that lack these qualities.

**Do**
- Take safeguarding of users funds seriously
- Provide suitable private key management schemes for beginners
- Offer progressive security and upgrade paths
- Build with bad actors in mind
- Minimize risk of self-inflicted loss from user negligence
- Continuously educate users on best practice and risks 
- Reduce attack surface by minimizing use of external code dependencies

**Don't**
- Blame the user for losing funds
- Expect beginners to implement best practice backup strategies
- Underestimate the added *feeling* of security that can come from well polished products

---

## Privacy

> Minimize collection of personal information, and maximize financial transaction privacy

> A common misconception of Bitcoin is that it provides complete anonymity and privacy of transactions. Since the blockchain is an unchangeable ledger of all transactions ever made, it is in fact very hard to not have your complete transaction history visible once even a single one of your addresses is connected to you. If Bitcoin is to become viable for a wider audience and daily use we ought to take privacy seriously. This is certainly not to enable or encourage illicit activity, but to protect individual financial privacy. We would not accept our bank to publish our financial transactions publicly.

Reviewer 3: I think an example might help to drive home the point here, "What if every time you spent or received cash, all the transaction details were published to your Twitter or Facebook feed for all your friends to see?" to use the example from [bitcoin.org](https://bitcoin.org/en/bitcoin-core/features/privacy). Or if you used bitcoin to pay for your Netflix subscription would you want them to be able to track past and future transactions?

I think a note on how privacy is a vital component underpinning financial freedom and freedom more generally. There is no freedom absent privacy in my opinion.

Author: Tweaked example in 7d0ef6e.

---
layout: guide
title: Principles
description: The key principles to follow when designing Bitcoin products.
nav_order: 9
parent: Foundations
permalink: /guide/foundations/principles/
main_classes: -no-top-padding
image: /assets/images/guide/foundations/principles/page-principles.jpg
---

<!--

Editor's notes

The key principles of designing for Bitcoin

-->

{% include picture.html
   image = "/assets/images/guide/foundations/principles/principles.jpg"
   retina = "/assets/images/guide/foundations/principles/principles@2x.jpg"
   mobile = "/assets/images/guide/foundations/principles/principles-mobile.jpg"
   mobileRetina = "/assets/images/foundations/principles/principles-mobile@2x.jpg"

>    alt-text = "Principles header illustration"

Reviewer: Alt text should describe the image content (which is a bit hard in this case due to the minimalism).

Author: Improved

   width = 1600
   height = 600
   layout = "full-width"
%}

# Bitcoin design principles

> As a new technology, Bitcoin offers the opportunity of a decentralized open financial system, where participants share the role of securing the network. This is important to give everyone equal and direct access without needing intermediaries. To make this a reality, we encourage everyone working on products to deliberately support the core principles of designing for Bitcoin. 

Reviewer: Is it important to say why not having intermediaries is something to strive for (like unfairly excluding certain people from being economically active, unfairly messing with peoples savings, etc)?

Author: Expanded.

- [Self-custody](#self-custody)
- [Security](#security)
- [Inclusion](#inclusion)
- [Interoperability](#interoperability)
- [Transparency](#transparency)
- [Privacy](#privacy)
- [Decentralization](#decentralization)

---

## Self-custody

> Let users control their private keys, with no risk for seizure or freezing of funds

Our existing mental models of access to digital services are usernames and passwords, controlled by a company that has custody of your funds and data. With everyone having direct access to the Bitcoin network without middlemen, there is no longer a need for people to delegate control of their funds. While it comes with greater responsibility, it is self-custody that enables the open financial system of peer-to-peer transactions.

**Do**
- Let users control their private keys and funds directly
- Create an easy path to self-custody for Bitcoin beginners

**Don't**

> - Custody funds for your users

Reviewer: ```suggestion
- Custody funds for your users unless it cannot be avoided for your specific use case, and you have your users full consent
```

Author: Do you have examples of where this might be required? (other than centralized exchanges)
Not 100% opposed to including this but want to understand more.

- Build products where the users' funds can be seized, or frozen

---

## Security

> Provide appropriate and progressive security for all types of users

Self-custody often leaves the end-user responsible for the security of their private keys. They can only do a good job of that if we provide them with appropriate tools and awareness of best practices. 

> Security is especially important when onboarding people new to Bitcoin where solutions that work for advanced users might backfire. New users are likely to start by storing small amounts until they get comfortable with self-custody. The concept of progressive security is a good idea for here. This would let a user upgrade their security and private key management scheme as their savings grow. 

Reviewer: ```suggestion
Security is especially important when onboarding people new to Bitcoin where solutions that work for advanced users might backfire. New users are likely to start by storing small amounts until they get comfortable with self-custody. The concept of progressive security is a good idea here. This would let a user upgrade their security and private key management scheme as their savings grow. 
```

> Education and awareness is a big part of security, protecting users from bad actors and potentially their own bad behavior. It is unrealistic to expect beginners to take in all the knowledge acquired by advanced users in one go, for example while [onboarding]({{ '/guide/onboarding/introduction/' | relative_url }}) to a bitcoin product. We should therefore consider how to continuously educate and level up user awareness of best practices and risks.

Reviewer: ```suggestion
Education and awareness is a big part of security, protecting users from bad actors and potentially their own mistakes. It is unrealistic to expect beginners to take in all the knowledge acquired by advanced users in one go, for example, while [onboarding]({{ '/guide/onboarding/introduction/' | relative_url }}) to a Bitcoin product. We should therefore consider how to continuously educate and level up user awareness of best practices and risks.
```

> Security can also be a feeling. A product that is polished, good looking, easy to use and transparently communicates how it works can help users feel more secure. Especialy compared to one with the exact same security measures that lack these qualities.

Reviewer: ```suggestion
Security can also be a feeling. A product that is polished, good looking, easy to use and transparently communicates how it works can help users feel more secure. Especially compared to one with the exact same security measures that lack these qualities.
```

**Do**

> - Take safeguarding of users funds seriously

Reviewer: ```suggestion
- Take safeguarding of user funds seriously
```

Author: I think it should either be,
- ... users' funds seriously
- ... user funds seriously

Author: Resolved to ... users' funds seriously, for now

- Strive for no loss of funds, whether by negligence or theft 
- Provide suitable private key management schemes for beginners
- Offer progressive security and upgrade paths
- Build with bad actors in mind
- Minimize risk of self-inflicted loss from user negligence
- Continuously educate users on best practice and risks 

> - Reduce attack surface by minimizing use of external code dependencies

Reviewer: ```suggestion
- Reduce attack surfaces by minimizing use of external code dependencies
```

**Don't**
- Blame the user for losing funds
- Expect beginners to implement best practice backup strategies
- Underestimate the added *feeling* of security that can come from well polished products

---

## Inclusion

> Build borderless products without location, language or social barriers

There are no background checks, credit checks, or gatekeepers to Bitcoin. A Kenyan farmer has the same access to Bitcoin as a Wall Street trader. 

While Bitcoin is already used by a large number of people, it pales in comparison with the many more that are likely to use it in the future. We need to design products that are prepared for people unfamiliar with Bitcoin. This means using plain and familiar language, explaining things in the context where they are needed, not overwhelming people with technical detail, and more.

**Do**
- Provide equal and direct access to the Bitcoin network
- Design Bitcoin products that are usable by the widest range of people possible
- Use plain language that people new to Bitcoin can understand regardless of prior knowledge
- Localize your product and make it multilingual 
- Educate in place, when people are presented with a new concept

> - Treat users who rely on assistive technologies, as first-class citizens

Reviewer: ```suggestion
- Treat users who rely on assistive technologies as first-class citizens
```

**Don't**
- Exclude people by building features that only work in certain countries

> - Add technical detail that is not required knowledge, or technical terms like seed phrase XPUBs, mnemonics etc.

Reviewer: ```suggestion
- Add technical detail that is not required knowledge, or technical terms like seed phrase, XPUBs, mnemonics etc.
```

- Put all education up front and expect people to read and remember it

---

## Interoperability

> Enable import and export of wallets, maximise backwards compatibility and use of open standards

Bitcoin is an open-source protocol, operated in a decentralized manner. This has led to a number of standards being developed for products building on the protocol to be compatible. It should be easy to switch and move your Bitcoin wallet to a different application, should you wish. Making sure that your product supports as many of these standards as possible is best practice and builds trust. More on [wallet interoperability]({{ '/guide/foundations/wallet-interoperability/' | relative_url }}). 

**Do**
- Support import and export of wallets
- Support as many relevant BIPs as possible
- Be transparent with which ones you do and don’t support
- Maximize backwards compatibility

**Don't**
- Lock your users in
- Implement proprietary solutions when open standards exist

---

## Transparency

> Be open and transparent with how your product works, open-source your code when possible

> While an open and decentralized financial system that lets people connect directly with it is great, it does put a lot of burden on the users to pick a product they trust, and that works for them. We can make this easier by freely sharing information about how our products work, what technologies they use and rely on. Let people verify that your claims are true, ideally by open-sourcing your own code. It is also very important to be open about what risks the user might be taking on with self-custody, under which scenarios they might lose access to their funds and best practices for avoiding this. 

Reviewer: ```suggestion
While an open and decentralized financial system that lets people connect directly with it has many benefits, it does put a lot of burden on the users to pick a product they trust, and that works for them. We can make this easier by freely sharing information about how our products work, what technologies they use and rely on. Let people verify that your claims are true, ideally by open-sourcing your own code. It is also very important to be open about what risks the user might be taking on with self-custody, under which scenarios they might lose access to their funds and best practices for avoiding this. 
```

**Do**
- Be open and transparent with how your product works
- Let people verify your claims by open-sourcing your code when possible
- Explain what risks the user is taking on, and how best to mitigate them

**Don't**
- Make claims that are not explained or verifiable 

---

## Privacy

> Minimize collection of personal information, and maximize financial transaction privacy

> A common misconception of Bitcoin is that it provides complete anonymity and privacy of transactions. Since the blockchain is an unchangeable ledger of all transactions ever made, it is in fact very hard to not have your complete transaction history visible once even a single one of your addresses is connected to you. If Bitcoin is to become viable for a wider audience and daily use we ought to take privacy seriously. This is certainly not to enable or encourage illicit activity, but to protect individual financial privacy. We would not accept our bank to publish our financial transactions to our Twitter or Facebook feeds, so should avoid that scenario with Bitcoin as well.

Reviewer: ```suggestion
A common misconception of Bitcoin is that it provides complete anonymity and privacy of transactions. Since the blockchain is an unchangeable ledger of all transactions ever made, it is in fact very hard to not have your complete transaction history visible once even a single one of your addresses is connected to you. If Bitcoin is to become viable for a wider audience and daily use we ought to take privacy seriously. This is certainly not to enable or encourage illicit activity, but to protect individual financial privacy. We would not accept our bank to publish our financial transactions to our Twitter or Facebook feeds, so we should avoid that scenario with Bitcoin as well.
```
